osgp/platform/osgp-adapter-ws-core/src/main/java/org/opensmartgridplatform/adapter/ws/core/application/mapping/DeviceConverterHelper.java

```diff
@@ -82,9 +83,10 @@ T initEntity(final Device source) {
                     source.getTechnicalInstallationDate().toGregorianCalendar().getTime());
         }
 
-        if (source.getDeviceModel() != null) {
-            final DeviceModel deviceModel = new DeviceModel();
-            deviceModel.setModelCode(source.getDeviceModel().getModelCode());
+        if (source instanceof Device) {
+            final Device sourceDevice = (Device) source;
+            destination.setDeviceModel(this.mapper.map(sourceDevice.getDeviceModel(),
```

Comment:
This seems like an improvement of the previous code, where it looks like the deviceModel is never set on the destination Device object. Could this potentially lead to unexpected side effects?